expose output service and more editor primitives

CompuIves opened the pull request:
There's also a `.codesandbox/Dockerfile` in there, which I used to make it work with CodeSandbox. I'm also happy to remove it if you prefer!

This PR exposes `IOutputService` and many more interfaces from the editor, in case you want to create your own `IEditorGroupsService`. I was contemplating whether I should just do `export * from '...'`, but went with the safe route for now :).

Fixes #138

CGNonofr commented:
Thank you!

Can you please follow the semantic release convention for your commits? (and splitting it in 2 commits - types and dockerfile) to make the automatic changelog relevant

How does it work for codesandbox though? I'm not familiar with it

CompuIves commented:
Okay! It's now split up in two commits.

> How does it work for codesandbox though? I'm not familiar with it

I've now added a [devcontainer](containers.dev) configuration for this repo. This essentially tells what environment this repo would need to properly run, which is useful for local development (VSCode can hint suggestions because of it), but it also helps for online development.

CodeSandbox is one of those online editors (I'm one of the co-founders, so there's bias). It allows you to open the repo in an online code editor, which already runs the repo so you don't have to clone locally and set up the environment yourself. Here's an example: https://codesandbox.io/p/github/CodinGame/monaco-vscode-api/main, although you can see it has an error currently because it lacks the configuration this PR contributes 😄 .

There are two advantages to it:

1. Every PR has a running dev env that you can use for reviewing, so you don't have to check it out locally
2. Contributors can open the online editor, click "Create Branch" and contribute more easily back to the repo

I've changed the CodeSandbox configuration to become a devcontainer configuration, so it will work with all online editors that use it (Codespaces as well, for example).

CGNonofr commented:
Ok, looks great! thanks!

CGNonofr commented:
https://codesandbox.io/p/github/CodinGame/monaco-vscode-api/main seems to still be not working, jq is required :)
